libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.java

```diff
@@ -495,6 +496,22 @@ public Builder addApproaches(String... approaches) {
       return this;
     }
 
+
+    /**
+     * Optionally, set which input coordinates should be treated as waypoints.
+     * <p>
+     * Most useful in combination with <tt>steps=true</tt> and requests based on traces
+     * with high sample rates. Can be an index corresponding to any of the input coordinates,
+     * but must contain the first ( 0 ) and last coordinates' index separated by <tt>;</tt>.
+     *
+     * @param waypoints integer array of coordinate indices to be used as waypoints
+     * @return this builder for chaining options together
+     */
+    public Builder addWaypoints(@Nullable @IntRange(from = 0) Integer... waypoints) {
```

Comment:
It may be better to be more specific with our naming here, as we already have a `addWaypoint`.  Maybe `addSlientWaypoint` or `addViaWaypoint`?

Reply:
Yeah, I couldn't agree more. Will fix 🚀